daweb/js2/sablony-komponenty/komponenty.md

```diff
@@ -0,0 +1,107 @@
+## Komponenty
+
+Abychom dokázali stavět větší aplikace a vyznali se v jejich kódu, bude se nám hodit rozdělit si vytváření obsahu stránky do menších funkcí. Představme si například hlavičku stránky s odkazy. Můžeme ji do stránky samozřejmě vložit takto:
+
+```jsx
+const appElement = document.querySelector('#app');
+appElement.innerHTML += render(
+  <header>
+    <a href="/">Domů</a>
+    <a href="/nakupy">Nákupy</a>
+    <a href="/kalendar">Kalendář</a>
+  </header>
+);
+```
+
+Do budoucna se nám však vyplatí vytvořit si pro tuto hlavičku samostatnou funkci. Nazveme ji `Header` a napíšeme ji tak, aby vracela JSX.
+
+```jsx
+const Header = () => {
+  return (
+    <header>
+      <a href="/">Domů</a>
+      <a href="/nakupy">Nákupy</a>
+      <a href="/kalendar">Kalendář</a>
+    </header>
+  );
+};
+```
+
+Tuto funkci pak můžeme použít při vytváření obsahu stránky.
+
+```jsx
+const appElement = document.querySelector('#app');
+appElement.innerHTML += render(Header());
```

Comment:
Tohle je kód, co si mají zapamatovat nebo co nejdříve zapomenout?

Reply:
Je to takový mezikrok, když ještě neumíme skládat komponenty. Jestli tě napdá, jak k tomu dojít bez toho, klidně tuhle věc nějak polaď podle sebe.